src/js/controller/model.ts

```diff
@@ -6,9 +6,10 @@ import { isValidNumber, isNumber } from 'utils/underscore';
 import { seconds } from 'utils/strings';
 import Providers from 'providers/providers';
 import type { StreamType } from 'providers/utils/stream-type';
-import type { DefaultProvider, GenericObject, TextTrackLike, Localization } from 'types/generic.type';
+import type { GenericObject, TextTrackLike, Localization } from 'types/generic.type';
 import type { QualityLevel } from 'providers/data-normalizer';
 import type PlaylistItem from 'playlist/item';
+import type { ProviderWithMixins } from 'providers/default';
```

Comment:
An external provider would not be a `ProviderWithMixins` but at runtime we will be dealing with external providers that do not use our mixins, so this is confusing.